workflow/workflow.go

```diff
@@ -607,6 +623,22 @@ func SetCurrentDetails(ctx Context, details string) {
 	internal.SetCurrentDetails(ctx, details)
 }
 
+// SetVersioningBehavior sets the strategy to upgrade this workflow when the default Build ID
+// has changed, and Worker Versioning-3 has been enabled.
+//
+// SetVersioningBehavior should be called during the first workflow task, and the context `ctx` should
+// be the original context workflow argument, or derived from this context.
+//
+// If not set, a default behavior provided in worker.Options.DefaultVersioningBehavior will be used.
+//
+// If not set, and the default behavior is also unspecified in the Worker, the first task of this workflow
+// will fail when Worker Versioning-3 has been enabled.
+//
+// NOTE: Experimental
+func SetVersioningBehavior(ctx Context, behavior VersioningBehavior) {
```

Comment:
I do not believe this is needed. Just use the worker option. I don't think we need global default for this worker option any more than any other (if a user wants to reference a global `EnableLoggingInReplay` or `VersioningBehavior` or whatever when they make their workers they can). This will because especially true as you get to other SDKs where we don't like customizable global defaults.

Reply:
Also agree

Reply:
This is not global, it only changes the running workflow options, so that we can override  the default for a particular workflow. I agree about no process-level defaults...

Reply:
I see, my mistake. I see this accepts a workflow context. I misread, sorry.